File: src/main/java/br/ufu/facom/petsi/controlaPET/model/Movement.java
package br.ufu.facom.petsi.controlaPET.model;

import br.ufu.facom.petsi.controlaPET.model.enums.MovementType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;

@Entity
@Table(name = "movements")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Movement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MovementType movementType;

    @Column(nullable = false)
    private int quantity;

    String notes;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDate movementDate;
}

File: frontend/src/interfaces/Movement.ts
export type MovementType = 'INBOUND' | 'OUTBOUND_CONSUMPTION' | 'ADJUSTMENT'

export interface Movement {
    id: number,
    userName: string,
    itemId: number,
    itemName: string,
    type: MovementType,
    notes: string,
    quantity: number,
    movementDate: string
}

export interface CreateMovementRequest {
    itemId: number,
    quantity: number,
    notes: String,
    type: MovementType
}

export interface ConsumeItem {
    itemId: number,
    quantity: number
}